Movpe 1 and movpe 2 merging + some solution class

aalbino2 opened the pull request.

aalbino2 commented:
Hey @hampusnasstrom if you can still see this email in the ocean that arrived you in the last hour, could you check this for me? Do not be concerned with movpe specific changes, I just need your comments on the init.py file in the root on the ikz_plugin 

Thanks!

hampusnasstrom commented on IKZ_plugin/src/ikz_plugin/__init__.py:
```diff
@@ -61,6 +72,282 @@ class SubstratePreparationStep(Activity):
+    )
+    intended_sample_conc = Quantity(
+        type=np.float64,
+        description="FILL THE DESCRIPTION",
```
Fill the description

aalbino2 replied:
done

hampusnasstrom commented on IKZ_plugin/src/ikz_plugin/__init__.py:
```diff
@@ -61,6 +72,282 @@ class SubstratePreparationStep(Activity):
+        type=str,
+        a_eln={"component": "StringEditQuantity"},
+    )
+    intended_sample_conc = Quantity(
```
spell out

hampusnasstrom replied:
Also, concentration of what component?

aalbino2 replied:
solved, placing it in the ComponentConcentration class

hampusnasstrom commented on IKZ_plugin/src/ikz_plugin/__init__.py:
```diff
@@ -61,6 +72,282 @@ class SubstratePreparationStep(Activity):
+    )
+    obtained_conc = Quantity(
+        type=np.float64,
+        description="FILL THE DESCRIPTION",
```
fill

aalbino2 replied:
done

hampusnasstrom commented on IKZ_plugin/src/ikz_plugin/__init__.py:
```diff
@@ -61,6 +72,282 @@ class SubstratePreparationStep(Activity):
+        a_eln={"component": "NumberEditQuantity", "defaultDisplayUnit": "mol / liter"},
+        unit="mol / liter",
+    )
+    obtained_conc = Quantity(
```
spell out

hampusnasstrom replied:
Again, of what component

aalbino2 replied:
I changed a bit it now: we have a repeated sub_section called component concentration where all these things are contained. The alias will sew together the component Quantified before and Mixed in here.

A solution reference exist now in the SolutionPreparationIKZ class, too

hampusnasstrom commented on IKZ_plugin/src/ikz_plugin/__init__.py:
```diff
@@ -61,6 +72,282 @@ class SubstratePreparationStep(Activity):
+    intended_sample_conc = Quantity(
+        type=np.float64,
+        description="FILL THE DESCRIPTION",
+        a_eln={"component": "NumberEditQuantity", "defaultDisplayUnit": "mol / liter"},
```
add label

hampusnasstrom commented on IKZ_plugin/src/ikz_plugin/__init__.py:
```diff
@@ -61,6 +72,282 @@ class SubstratePreparationStep(Activity):
+    obtained_conc = Quantity(
+        type=np.float64,
+        description="FILL THE DESCRIPTION",
+        a_eln={"component": "NumberEditQuantity", "defaultDisplayUnit": "mol / liter"},
```
label

aalbino2 replied:
I moved this in the mixing. I think it is fine to have it there. Also, I made it as an array.

But I'm open to delete this quantity, too. It might be included at the single use case level, if needed

aalbino2 replied:
moved it in ComponentConcentration class

hampusnasstrom commented on IKZ_plugin/src/ikz_plugin/__init__.py:
```diff
@@ -61,6 +72,282 @@ class SubstratePreparationStep(Activity):
+        a_eln={"component": "NumberEditQuantity", "defaultDisplayUnit": "mol / liter"},
+        unit="mol / liter",
+    )
+    intended_tot_volume = Quantity(
```
spell out

hampusnasstrom commented on IKZ_plugin/src/ikz_plugin/__init__.py:
```diff
@@ -61,6 +72,282 @@ class SubstratePreparationStep(Activity):
+    )
+    intended_tot_volume = Quantity(
+        type=np.float64,
+        description="FILL THE DESCRIPTION",
```
fill

hampusnasstrom commented on IKZ_plugin/src/ikz_plugin/__init__.py:
```diff
@@ -61,6 +72,282 @@ class SubstratePreparationStep(Activity):
+    intended_tot_volume = Quantity(
+        type=np.float64,
+        description="FILL THE DESCRIPTION",
+        a_eln={"component": "NumberEditQuantity", "defaultDisplayUnit": "mol / liter"},
```
label

hampusnasstrom replied:
correct unit

aalbino2 replied:
true

hampusnasstrom commented on IKZ_plugin/src/ikz_plugin/__init__.py:
```diff
@@ -61,6 +72,282 @@ class SubstratePreparationStep(Activity):
+        type=np.float64,
+        description="FILL THE DESCRIPTION",
+        a_eln={"component": "NumberEditQuantity", "defaultDisplayUnit": "mol / liter"},
+        unit="mol / liter",
```
correct unit

hampusnasstrom commented on IKZ_plugin/src/ikz_plugin/__init__.py:
```diff
@@ -61,6 +72,282 @@ class SubstratePreparationStep(Activity):
+        a_eln={"component": "NumberEditQuantity", "defaultDisplayUnit": "mol / liter"},
+        unit="mol / liter",
+    )
+    obtained_tot_volume = Quantity(
+        type=np.float64,
+        description="FILL THE DESCRIPTION",
+        a_eln={"component": "NumberEditQuantity", "defaultDisplayUnit": "mol / liter"},
+        unit="mol / liter",
```
see above

hampusnasstrom commented on IKZ_plugin/src/ikz_plugin/__init__.py:
```diff
@@ -61,6 +72,282 @@ class SubstratePreparationStep(Activity):
+
+    temperature = Quantity(
+        type=np.dtype(np.float64),
+        unit=("°C"),
```
SI?

aalbino2 replied:
done

hampusnasstrom commented on IKZ_plugin/src/ikz_plugin/utils.py:
```diff
@@ -102,3 +103,52 @@ def create_archive(
     )
+
+
+def create_timeseries_objects(
```
add docstring

aalbino2 replied:
I moved this function into movpe1 folder

hampusnasstrom commented on IKZ_plugin/src/ikz_plugin/utils.py:
```diff
@@ -102,3 +103,52 @@ def create_archive(
+    return objects
+
+
+def row_to_array(dataframe: pd.DataFrame, quantities, row_index):
```
add docstring

aalbino2 replied:
improved ;)

hampusnasstrom commented on IKZ_plugin/src/ikz_plugin/__init__.py:
```diff
@@ -61,6 +72,282 @@ class SubstratePreparationStep(Activity):
+    )
+
+    steps = SubSection(
+        section_def=SolutionPreparationStep,
```
So now we can't nest solution preparations but also dropped the alias?

aalbino2 replied:
true, I will restore the aliases and drop the nesting solution thing

hampusnasstrom commented on IKZ_plugin/src/ikz_plugin/movpe/schema.py:
The git diff of this file is a bit crazy so it's quite hard to review. What is happening in here?

aalbino2 replied:
I changed and improved the data model, I'm also testing it so it is fine not to go in detail on this movpe.schema.py file

hampusnasstrom reviewed (changes requested):
I think we need to test and discuss the solution prep a bit. Maybe it's okay to have it like this for now if you are okay with updating all the MOVPE archives ones we have a final version. The movpe.schema diff was quite confusing, what is the intended change in this file?

aalbino2 commented:
Done! I tried to solve the doubts on what intended_conc and obtained_conc were standing for.

I hope you find it better now! Thanks for comments, I think it improved

Let'S see if we find a point to merge tomorrow. Then, if we need to chnage it further, I will be avail to update the upload of the movpe people anyway

RoteKekse commented:
Hey some feedback from me:
The quantities `temperature` `time` and `speed` are solution preparation steps. In my classes i hasd them on toplevel and kept them there for backwards compatibility but moved them in an extra section.

The term `final_concentration` is a bit difficult in context where more then on substance is desolved and it is not clear by context to what this referes.

I think the `component` quantity is missing other measures of amount besides `mass`. solvents usually have a volume. sometimes the solution is only describe by relative values, such as volume or mass or molar concentrations.

aalbino2 commented:
Thanks for comments, indeed I think you make good points here and I refined the classes in the following way

- I removed temperature, time, and speed from Solution class
- I changed this final concentration thing, now there is the ComponentConcentration repeated class referencing a system and specifing its concentrantion, the same class is also used within the mixing step.
- You're right about component, indeed elsewhere I had this liquidComponent subclass, that now I added here
- I removed additive, that was a leftover from your classes still.

hampusnasstrom reviewed (approved):
LGTM
